Adds unit test for mpp_root_pe()

mlee03 opened the pull request:
**Description**
Adds unit test for the function mpp_root_pe().

Fixes # (issue)

**How Has This Been Tested?**
Make check runs successfully on Gaea and Skylake.
Make distcheck runs successfully on Skylake.
Passes Travis-CI

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

colingladueNOAA commented on test_fms/mpp/test_mpp_root_pe.sh:
```diff
@@ -0,0 +1,39 @@
+if [ $(command -v nproc) ] ; then
+  nProc=$(nproc)
+  nProc=$( echo $nProc/2 | bc )
+  run_test test_mpp_npes ${nproc}
```
needs to be test_mpp_root_pe

mlee03 replied:
@colingladueNOAA,  I changed the doxygen section to follow the guidelines in CODE_STYLE document.  I also added an additional run_test line in test_mpp_root_pe.sh to run with nproc number of processors, where nproc is a user-defined variable set to 4 as default.

mlee03 commented:
@colingladueNOAA, I changed the doxygen section of the code to follow the guidelines in the CODE_STYLE document.  I also added another test_run in test_mpp_root_pe.sh to run with nproc number of processors, where nproc is a user-defined variable that is set to 4 as default.
